### PR TITLE
add support for configurable build dir when building nuget packages

### PR DESCRIPTION
**Pull request by askhade**
**Description**: ORT build let's to choose build dir by specifying --build_dir arg. Nuget package creation does not honor this and tries to always build the nuget packages using the default build directory. This results in packaging wrong native libs and sometimes package creation failures (depending on whether the default build dir exists or not). This PR fixes this issue.

**Motivation and Context**
- Fixes a bug
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: tools/ci_build/build.py [hunk 1]
```diff
@@ -1467,33 +1467,38 @@ def build_nuget_package(configs, use_cuda, use_openvino, use_tensorrt, use_dnnl,
     else:
         pass
 
+    # set build directory based on build_dir arg
+    native_dir = os.path.normpath(os.path.join(source_dir, build_dir))
```
Comment (edgchen1) [inline tools/ci_build/build.py:1471]: why is build_dir appended to source_dir? what if build_dir is an absolute path?

File: tools/ci_build/build.py [hunk 2]
```diff
@@ -1467,33 +1467,38 @@ def build_nuget_package(configs, use_cuda, use_openvino, use_tensorrt, use_dnnl,
...
+    OnnxRuntimeBuildDirectory = "/p:OnnxRuntimeBuildDirectory=\"" + native_dir + "\""
```
Comment (edgchen1) [inline tools/ci_build/build.py:1472]: nit: inconsistent variable naming

File: tools/ci_build/build.py (hunk 1 above)
Comment (askhade) [reply tools/ci_build/build.py:1471]: If a component is an absolute path, all previous components are thrown away and joining continues from the absolute path component.

**Review by pranavsharma — APPROVED** [review]
Thanks!

**Pull request merged**